contrib/core-contract-tests/tests/pox-4/pox-4.stateful-prop.test.ts

```diff
@@ -0,0 +1,84 @@
+import { describe, it } from "vitest";
+
+import { initSimnet } from "@hirosystems/clarinet-sdk";
+import { Real, Stub } from "./pox_CommandModel.ts";
+
+import {
+  getPublicKeyFromPrivate,
+  publicKeyToBtcAddress,
+} from "@stacks/encryption";
+import { StacksDevnet } from "@stacks/network";
+import {
+  createStacksPrivateKey,
+  getAddressFromPrivateKey,
+  TransactionVersion,
+} from "@stacks/transactions";
+import { StackingClient } from "@stacks/stacking";
+
+import fc from "fast-check";
+import { PoxCommands } from "./pox_Commands.ts";
+
+describe("PoX-4 invariant tests", () => {
+  it("statefully does solo stacking with a signature", async () => {
+    // SUT stands for "System Under Test".
+    const sut: Real = {
+      network: await initSimnet(),
+      wallets: [
+        "7287ba251d44a4d3fd9276c88ce34c5c52a038955511cccaf77e61068649c17801",
+        "d655b2523bcd65e34889725c73064feb17ceb796831c0e111ba1a552b0f31b3901",
+        "3eccc5dac8056590432db6a35d52b9896876a3d5cbdea53b72400bc9c2099fe801",
+        "7036b29cb5e235e5fd9b09ae3e8eec4404e44906814d5d01cbca968a60ed4bfb01",
+        "b463f0df6c05d2f156393eee73f8016c5372caa0e9e29a901bb7171d90dc4f1401",
+      ].map((prvKey) => {
+        const pubKey = getPublicKeyFromPrivate(prvKey);
+        const devnet = new StacksDevnet();
+        const signerPrvKey = createStacksPrivateKey(prvKey);
+        const signerPubKey = getPublicKeyFromPrivate(signerPrvKey.data);
+        const btcAddress = publicKeyToBtcAddress(pubKey);
+        const stxAddress = getAddressFromPrivateKey(
+          prvKey,
+          TransactionVersion.Testnet,
+        );
+
+        return {
+          prvKey,
+          pubKey,
+          stxAddress,
+          btcAddress,
+          signerPrvKey: signerPrvKey,
+          signerPubKey: signerPubKey,
+          client: new StackingClient(stxAddress, {
```

Comment:
@hugocaillard this was compiling with `"@hirosystems/clarinet-sdk": "^2.4.0-beta2` but it looks like the type constructor signature has changed(?)

This branch is based on `next` which uses `"@hirosystems/clarinet-sdk": "^2.4.0-beta4`, in case it helps to troubleshoot.

Reply:
@hugocaillard, I think I see where the issue is. The PoX-4 stacking client is on a different [branch](https://github.com/hirosystems/stacks.js/compare/main...feat/update-stacking-client-to-pox-4). I got the tests passing now by referencing `@stacks/stacking` [6.12.1-pr.2cf6997.0](https://www.npmjs.com/package/@stacks/stacking/v/6.12.1-pr.2cf6997.0) in https://github.com/stacks-network/stacks-core/pull/4550/commits/66b9bd978ae9c73dc4c2fe380f164bb3c2d04e86.